### DISCUSSION TITLE
Force Table Reload after Action on BelongsToManyRelationManager

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#1810] [Help]
I am adding a custom action via the `pushActions()` method on the table of a `BelongsToManyRelationManager` class and need to force the entire table to refresh/reload after the action successfully completes. I have done something very similar using the `getTableActions()` method on a `ListRecords` class and was able to use `$this->getTableRecords()->fresh()`. However, when trying to use the same code in the `BelongsToManyRelationManager` class, it doesn't work. I get an error about using `$this` when not in object context.

My code:

```php
public static function table(Table $table): Table
{
    return $table
        ->columns([
          // the columns
        ])
        ->pushActions([
            Tables\Actions\LinkAction::make('custom_action')
                ->label('Do it!')
                ->action(function (Fixture $record) {
                    // perform custom action
        
                    $this->notify('success', 'It worked!');
        
                    $this->getTableRecords()->fresh();
                }),
        ]);
}
```

**Comment by studiowizjo** [comment]
Try this:

```php
->pushActions([
Tables\Actions\LinkAction::make('custom_action')
    ->label('Do it!')
    ->action(function ($record, $livewire) {
        // perform custom action

        \Filament\Facades\Filament::notify('success', 'It worked!');

        $livewire->getTableRecords()->fresh();
    }),
]);
```

**Reply by intrepidws** [reply]
Thanks @studiowizjo! I actually had to change the way my code worked entirely, for other reasons - so I can't confirm whether this works or not. But given how often I search this repo, and end up on one of my own issues, I'm sure this will come in handy.
